### PR TITLE
Feature/111 custom dimensions with action scope

### PR DESCRIPTION
**Pull request by brototyp**
Implements #111 for the action scope.

Basically I made the `CustomDimension` struct public and added an optional parameter to the tracking functions to add additional dimensions.

Since the `CustomDimension` is a struct, it is not usable from Objective-C. Still I think this is the best idea.

### CODE REVIEW

File: PiwikTracker/PiwikTracker.swift [hunk 1]
```diff
@@ -260,13 +216,15 @@ extension PiwikTracker {
     ///
     /// - Parameter view: An array of hierarchical screen names.
     /// - Parameter url: The url of the page that was viewed. If none set the url will be http://example.com appended by the screen segments. Example: http://example.com/players/john-appleseed
```
Comment (thorstenstark) [inline PiwikTracker/PiwikTracker.swift]: The documentation here should mention the `dimensions` parameter.

File: PiwikTracker/PiwikTracker.swift [hunk 2]
```diff
@@ -260,13 +216,15 @@ extension PiwikTracker {
...
-    @objc public func track(view: [String], url: URL? = nil) {
-        queue(event: event(action: view, url: url))
+    public func track(view: [String], url: URL? = nil, dimensions: [CustomDimension] = []) {
```
Comment (thorstenstark) [inline PiwikTracker/PiwikTracker.swift]: As `CustomDimension` isn't visible to objc, I'm not sure if these methods are still visible to objc.

File: PiwikTracker/PiwikTracker.swift (hunk 1 above)
Comment (brototyp) [reply PiwikTracker/PiwikTracker.swift]: Good catch! Done

File: PiwikTracker/PiwikTracker.swift (hunk 2 above)
Comment (brototyp) [reply PiwikTracker/PiwikTracker.swift]: There is a Objective-C extension without the dimensions property. See https://github.com/piwik/piwik-sdk-ios/pull/193/files#diff-bd7b3c5478dc6552eb40fd9baaa55d18R289

**Pull request merged**